compiler/frontend/src/org/jetbrains/kotlin/types/expressions/ExpressionTypingVisitorForStatements.java

```diff
@@ -477,6 +486,41 @@ protected KotlinTypeInfo visitAssignment(KtBinaryExpression expression, Expressi
         return resultInfo.replaceType(components.dataFlowAnalyzer.checkStatementType(expression, contextWithExpectedType));
     }
 
+    private KotlinTypeInfo resolveAssignOperatorOverload(
+            BindingContext bindingContext,
+            KtBinaryExpression expression,
+            ExpressionTypingContext contextWithExpectedType,
+            KtExpression leftOperand,
+            KtExpression left,
+            KotlinTypeInfo leftInfo,
+            ExpressionTypingContext context
+    ) {
+        KotlinType leftType = leftInfo.getType();
+        KotlinType assignmentOperationType = null;
+        TemporaryTraceAndCache temporaryForAssignmentOperation = null;
+        VariableDescriptor descriptor = BindingContextUtils.extractVariableFromResolvedCall(bindingContext, leftOperand);
+        OverloadResolutionResults<FunctionDescriptor> assignmentOperationDescriptors = new NameNotFoundResolutionResult<>();
+        if (descriptor != null && !descriptor.isVar()) {
```

Comment:
Should we just do early return in case `descriptor.isVar() == true`?

Reply:
Makes sense.